feat: Allow traefik merged dynamic configuration override with <project>.yaml (#6375)
ddev/ddev: docs/content/users/extend/traefik-router.md

All Traefik configuration uses the _file_ provider, not the _Docker_ provider. Even though the Traefik daemon itself is running inside the `ddev-router` container, it uses mounted files for configuration, rather than listening to the Docker socket.

!!!tip
Like other DDEV configuration, any file with `#ddev-generated` will be overwritten unless you choose to “take over” it yourself. You can do this by removing the `#ddev-generated` line. DDEV will stop making changes to that file and you’ll be responsible for updating it.

### Traefik Static Configuration

Static configuration is automatically generated in the `~/.ddev/traefik` directory. "Static" configuration means Traefik configuration which is only read when the router is started.

- `.static_config.yaml` (a hidden file) is the configuration that gets used. It is not to be edited; it is generated from DDEV's base configuration while merging any files named `static_config.*.yaml`. It is read on router startup, and does not change until the router starts again (normally after `ddev poweroff`).

> - Additional static configuration may be added by adding `static_config.*.yaml` files, which will be merged into the generated `.static_config.yaml`. For example, a `static_config.plugin.yaml` might contain external Traefik plugins, or a `static_config.dnschallenge.yaml` might provide configuration for additional `certificatesResolvers`. Merging is done with an **override** strategy, meaning that the final file in alphanumeric sort to touch a particular element of the YAML structure wins.

Reviewer: One more `plugin` > `plugins`:

```suggestion
- Additional static configuration may be added by adding `static_config.*.yaml` files, which will be merged into the generated `.static_config.yaml`. For example, a `static_config.plugins.yaml` might contain external Traefik plugins, or a `static_config.dnschallenge.yaml` might provide configuration for additional `certificatesResolvers`. Merging is done with an **override** strategy, meaning that the final file in alphanumeric sort to touch a particular element of the YAML structure wins.
```

* The `certs` directory contains the `<projectname>.crt` and `<projectname>.key` certificate generated for the project.
* The `config/<projectname>.yaml` file contains the configuration for the project, including information about routers, services, and certificates.
    * You can modify a project's dynamic configuration by adding `dynamic_config.*.yaml` files to `.ddev/traefik/`. These will be merged with an alphanumeric sort **override** strategy, so anything that is needed can be added or replaced.
    * An example `.ddev/traefik/dynamic_config.middlewares.yaml.example` file is provided in `.ddev/traefik/`. It can be modified to define and assign different middlewares, as well as overriding any default parameters. Please see `.ddev/traefik/README.md` file for more details.

>     * The extra config files may contain Go template placeholders, for injecting App-specific data into the config. But this is an advanced option that most can ignore. If you use them, note that template `{{ }}` placeholders cannot go at the start of a key (e.g. `{{.App.Name}}-web-80-http`). They must be preceded by any character, or used in YAML values. This means that if a middleware needs to be prefixed with the app's name, it will either need to be done manually or prefixed when an add-on creates its dynamic_config.*.yaml file from its own go template.

Reviewer: ```suggestion
    * The extra config files may contain Go template placeholders, for injecting App-specific data into the config. But this is an advanced option that most can ignore. If you use them, note that template `{{ }}` placeholders cannot go at the start of a key (e.g. `{{.App.Name}}-web-80-http`). They must be preceded by any character, or used in YAML values. This means that if a middleware needs to be prefixed with the app's name, it will either need to be done manually or prefixed when an add-on creates its `dynamic_config.*.yaml` file from its own go template.
```

>     * The extra config files may contain Go template placeholders, for injecting App-specific data into the config. But this is an advanced option that most can ignore. If you use them, note that template `{{ }}` placeholders cannot go at the start of a key (e.g. `{{.App.Name}}-web-80-http`). They must be preceded by any character, or used in YAML values. This means that if a middleware needs to be prefixed with the app's name, it will either need to be done manually or prefixed when an add-on creates its `dynamic_config.*.yaml` file from its own go template.

Reviewer 2: Maybe this can go in the README (more obscure) but be left out of docs (here?)

* The `certs` directory contains the `<projectname>.crt` and `<projectname>.key` certificate generated for the project.
* The `config/<projectname>.yaml` file contains the configuration for the project, including information about routers, services, and certificates.

>     * If you alter a `config/<projectname>.yaml` file, you need to remove the `#ddev-generated` from the top of the file, so it won't be overwritten on the next `ddev start`.

Reviewer 2: Let's just not suggest this. People may be familiar with the technique and they'll know what to do, and it will work.
```suggestion
```

>     * However, as with extending the `static_config.yaml` file, the new way to modify a project's dynamic configuration is by adding `dynamic_config.*.yaml` files to `.ddev/traefik/`. These will be merged with the same alphanumeric sort **override** strategy.

Reviewer 2: ```suggestion
    * You can modify a project's dynamic configuration by adding `dynamic_config.*.yaml` files to `.ddev/traefik/`. These will be merged with an alphanumeric sort **override** strategy, so anything that is needed can be added or replaced.
```

>     * A `dynamic_config.middlewares.yaml.example` file is now created in `.ddev/traefik/` which can be modified to define and assign different middlewares, as well as overriding any default parameters. Please refer to the `.ddev/traefik/README.md` file for more details.

Reviewer 2: ```suggestion
    * An example `.ddev/traefik/dynamic_config.middlewares.yaml.example` file is provided in `.ddev/traefik/. It can be modified to define and assign different middlewares, as well as overriding any default parameters. Please see `.ddev/traefik/README.md` file for more details.
```
